File: trivially/src/main/java/trivially/ep/m6/finalized/Mult.java
package trivially.ep.m6.finalized;

import trivially.ep.m6.Exp;

public class Mult implements trivially.ep.m6.Mult {

    protected Exp left;

    public Exp getLeft() {
        return this.left;
    }

    protected Exp right;

    public Exp getRight() {
        return this.right;
    }

    // Warning: Unsafe casts
    public void setLeft(trivially.ep.m4.Exp left) { this.left = (trivially.ep.m6.Exp) left; }
    public void setRight(trivially.ep.m4.Exp right) { this.right = (trivially.ep.m6.Exp) right; }

    public Mult(Exp left, Exp right) {
        this.left = left;
        this.right = right;
    }
}

File: coco_csharp/CoCo/EP/M7/Finalized/Mult.cs
namespace EP.M7.Finalized {
    public class Mult : Exp, EP.M7.Mult<Exp> {
        private EP.Exp<Exp> left;
        private EP.Exp<Exp> right;

        public Mult(EP.Exp<Exp> _left, EP.Exp<Exp> _right) {
            this.left = _left;
            this.right = _right;
        }

        public EP.Exp<Exp> getLeft() { return this.left; }
        public void setLeft(EP.Exp<Exp> _left) { this.left = _left; }

        public EP.Exp<Exp> getRight() { return this.right; }
        public void setRight(EP.Exp<Exp> _right) { this.right = _right; }
    }
}